archunit/src/main/java/com/tngtech/archunit/library/dependencies/CycleConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2014-2020 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.library.dependencies;
+
+import com.tngtech.archunit.ArchConfiguration;
+
+final class CycleConfiguration {
+    static final String MAX_NUMBER_OF_CYCLES_TO_DETECT_PROPERTY_NAME = "cycles.maxNumberToDetect";
+    private static final String MAX_NUMBER_OF_CYCLES_TO_DETECT_DEFAULT_VALUE = "100";
+    static final String MAX_NUMBER_OF_DEPENDENCIES_TO_SHOW_PER_EDGE_PROPERTY_NAME = "cycles.maxNumberOfDependenciesPerEdge";
+    private static final String MAX_NUMBER_OF_DEPENDENCIES_TO_SHOW_PER_EDGE_DEFAULT_VALUE = "20";
```

Comment:
Same applies here.

Reply:
Same argument as above applies here :wink: Do I really want to read so many dependencies per edge? I would guess normally there is one direction that is intended, where there are a lot of dependencies, and then there is this wrong edge that creates the cycle, which maybe has only a couple of dependencies (yes, in legacy projects that's probably too optimistic :wink:)
I mean, this is only a default, if anyone doesn't like it it can be reconfigured. But remember that this is going to be the setting **for every edge**, so it quickly explodes.

Reply:
I agree. I didn't think human beeings as consumers of the test results.